draft-ietf-quic-transport.md

```diff
@@ -2061,8 +2069,9 @@ From this state, either endpoint can send a frame with the FIN flag set, which
 causes the stream to transition into one of the "half-closed" states.  An
 endpoint sending an FIN flag causes the stream state to become "half-closed
 (local)".  An endpoint receiving a FIN flag causes the stream state to become
-"half-closed (remote)"; the receiving endpoint MUST NOT process the FIN flag
-until all preceding data on the stream has been received.
+"half-closed (remote)" once all preceding data has arrived.  The receiving
+endpoint MUST NOT consider the stream state to have changed until all data has
+arrived.
```

Comment:
Interesting question: what does the sending peer assume?  I think that they have to assume that the state isn't closed until all data has been acknowledged (or their RST_STREAM is acknowledged).  (Oh, I see that you caught that, awesome).